Sources/EmbraceCore/Session/DataRecovery/UnsentDataHandler.swift

```diff
@@ -273,4 +274,43 @@ class UnsentDataHandler {
         let sessionId = currentSessionId ?? Embrace.client?.currentSessionId()
         storage.cleanMetadata(currentSessionId: sessionId, currentProcessId: ProcessIdentifier.current.hex)
     }
+
+    static func sendCriticalLogs(fileUrl: URL?, upload: EmbraceUpload?) {
+
+        guard let upload = upload,
+              let fileUrl = fileUrl else {
+            return
+        }
+
+        // always remove the logs from previous session
+        defer { try? FileManager.default.removeItem(at: fileUrl) }
+
+        guard let logs = try? String(contentsOf: fileUrl) else {
```

Comment:
Same as my comment above. Simply loading all the contents like this can get very intense on resources and cause memory issues. Do we have the facilities to upload a file on disk and let the OS do the grunt work?